NitroxClient/GameLogic/Cyclops.cs

```diff
@@ -53,5 +75,452 @@ public void ActivateShield(string guid)
+        {
+            BroadcastDamageChange(subRoot, info);
+        }
+
+        /// <summary>
+        /// Called when the player repairs a <see cref="CyclopsDamagePoint"/>. Right now it's not possible to partially repair because it would be difficult to implement.
+        /// <see cref="CyclopsDamagePoint"/>s are coupled with <see cref="LiveMixin"/>, which is used with just about anything that has health.
+        /// I would need to hook onto <see cref="LiveMixin.AddHealth(float)"/>, or maybe the repair gun event to catch when something repairs a damage point, which I don't 
+        /// believe is worth the effort. A <see cref="CyclopsDamagePoint"/> is already fully repaired in a little over a second. This can trigger sending 
+        /// <see cref="CyclopsDamagePointHealthChanged"/> and <see cref="CyclopsDamage"/> packets
+        /// </summary>
+        public void OnDamagePointHealthChanged(SubRoot subRoot, CyclopsDamagePoint damagePoint, float repairAmount)
+        {
+            string subGuid = GuidHelper.GetGuid(subRoot.gameObject);
+
+            for (int i = 0; i < subRoot.damageManager.damagePoints.Length; i++)
+            {
+                if (subRoot.damageManager.damagePoints[i] == damagePoint)
+                {
+                    CyclopsDamagePointHealthChanged packet = new CyclopsDamagePointHealthChanged(subGuid, i, repairAmount);
+                    packetSender.Send(packet);
+
+                    BroadcastDamageChange(subRoot, Optional<DamageInfo>.Empty());
+
+                    return;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Set the health of a <see cref="CyclopsDamagePoint"/>. This can trigger sending <see cref="CyclopsDamagePointHealthChanged"/>
+        /// and <see cref="CyclopsDamage"/> packets
+        /// </summary>
+        /// <param name="repairAmount">The max health of the point is 1. 999 is passed to trigger a full repair of the <see cref="CyclopsDamagePoint"/></param>
+        public void SetDamagePointHealth(SubRoot subRoot, int damagePointIndex, float repairAmount)
+        {
+            subRoot.damageManager.damagePoints[damagePointIndex].liveMixin.AddHealth(repairAmount);
+        }
+
+        /// <summary>
+        /// This is run during every major state change. If this player is the current owner, a <see cref="CyclopsDamage"/> packet will be sent out
+        /// </summary>
+        private void BroadcastDamageChange(SubRoot subRoot, Optional<DamageInfo> info)
+        {
+            string subGuid = GuidHelper.GetGuid(subRoot.gameObject);
+
+            // If it isn't the owner, we don't want to broadcast
+            if (!simulationOwnershipManager.HasAnyLockType(subGuid))
+            {
+                return;
+            }
+
+            LiveMixin subHealth = subRoot.gameObject.RequireComponent<LiveMixin>();
+
+            SerializableDamageInfo damageInfo = null;
+
+            if (info.IsPresent())
```

Comment:
Should this block be inside the below `if (subHealth.healh > 0)` block? (we don't want to compile this data if the cyclops ist going to be destroyed.)  If you were to do that, might make sense to have a new function to contain all of the logic like: `buildCyclopsDamagePacket()`